### ISSUE TITLE
Some biomes completely bare

### ISSUE THREAD

**Issue by ATLCode** [CorgiTaco/BYG#168]
**Before filling out this issue: **
**1. For major bugs, we only support BYG's latest version on a per Minecraft version basis(i.e latest for 1.15 is 1.0 while latest for 1.16 is 1.1).**

This is about 1.1.5 Version on Forge 1.16.4 - 35.0.15 (So might of course be something related to that)

**2. Please test BYG on its own to confirm whether or not this is a mod conflict**

Did so.

__**Mod conflict? If so, please clarify which mod is conflicting, if not skip this.**__

__**API/Modloader**__
Forge

__**Describe the bug**__
Some biomes lack trees etc. entirely. Wild guess is that "seasonal birch forest" for example is not supposed to be bare dirt land. I couldn't quickly find wiki etc. with all mod's biomes listed but here are quick summary on some:

Doesn't work:
-Seasonal Birch Forest
-Deciduous Forest
-Seasonal Forest
-Boreal Forest
-Zelkova Forest

Works:
-Blue Taiga
-Grove
-Mangrove Marches
-Ebony Woods
-Allium Fields
-Maple Taiga

__**To Reproduce**__
Load byg-1.1.5.jar with  Forge 1.16.4 - 35.0.15 and fly around.

**Expected behavior**
Some biomes are just as you expect them to and others are bare dirt/grass lands with no tree's etc. at all.

**Screenshots/Videos**
-

**Additional context**
Thought I'd just let you guys know since didn't notice anyone report it yet, even though it's expected.

**Logs**
Use: https://paste.ee/ and paste the links. if log is huge please upload those
Please provide:
Crashlog - 
Debug.log - 
Latest.log - 

**Heap Dumps**
*If performance, please provide a heap dump/some sort of profile of your game(I recommend: https://visualvm.github.io/download.html)*
Heap Dump/Profile -

**Comment by CorgiTaco** [comment]
You're not giving me much information to work with. I need at least the latest.log(this has your mod list)

**Comment by ATLCode** [comment]
My bad, didn't realize I had optifine on and low and behold. Optifine + BYG reproduces it, BYG alone works.

These should be of some help.

[2020-11-13-2.log.gz](https://github.com/CorgiTaco/BYG/files/5539223/2020-11-13-2.log.gz)
[debug-1.log.gz](https://github.com/CorgiTaco/BYG/files/5539224/debug-1.log.gz)

**Comment by CorgiTaco** [comment]
They released the fix for this yesterday so please download the latest Optifine for 1.16.4

**Comment by ATLCode** [comment]
I see, my bad. Sorry for the bother.

**Comment by ATLCode** [comment]
Where can one get this latest version though? As far as I've been able to find latest update is OptiFine HD U G5 released on 06/11, which I think is what I had when testing this.

**Comment by CorgiTaco** [comment]
read what you said again. "think" this is telling me, you're not using the latest optifine

**Comment by ATLCode** [comment]
I don't quite get it. Debug-1.log.gz says "OptiFine ZIP file: C:\Users\ATL\AppData\Roaming\.minecraft\mods\OptiFine_1.16.4_HD_U_G5.jar"

Which as far as I can figure out is the latest version available. Does it show up as something different to you or what am I missing.

I was assuming you were talking about there being new optifine version since you mentioned they released the fix yesterday 3 days ago. However at least here: https://optifine.net/downloads the newest releases (1.16.4 HD_U_G5) are/is almost almost 2 weeks old.

I'm just bit confused. I tried it again, making sure I had latest version of both optifine and BYG (at least latest I can find in the official pages for both on 1.16.4) and same issue seems to pop up. Just to check is this something you don't have happening if running latest version of BYG and optifine together?

[2020-11-14-2.log.gz](https://github.com/CorgiTaco/BYG/files/5550448/2020-11-14-2.log.gz)
[debug.log](https://github.com/CorgiTaco/BYG/files/5550449/debug.log)

**Comment by SatDog92** [comment]
@CorgiTaco 
>  you're not using the latest optifine

The same thing happened to me, though, with latest mod version (Biomes You Go 1.1.5) and latest Optifine version as well, download from the [Optifine website](https://optifine.net/downloads) (OptiFine HD U G5), released on November 6. First i used Forge 35.0.1 (cause it was the version Optifine claimed to be compatible of), then i tried the 35.0.17, but the problem seems to not go away (even if Optifine still works). [log files byg.zip](https://github.com/CorgiTaco/BYG/files/5555908/log.files.byg.zip)

You also mentioned the fix was released the day before, November 12, but on the website there is literally no update of it on November 12, only the  OptiFine HD U G5 released on November 6. ATLCode is right to be confused. I'm 100% sure you are not screwing with us (Edit: no really, it's not a provocation) and that there is some misunderstanding, are we missing a download link maybe? Because i downloaded all the mods' latest versions (except Forge 35.0.18, which had a bug symbol in front of the name, so i went for the 35.0.17 one).

It's a pity because, since according to you this bug was fixed already, if i cant fix this issue then i will have to enjoy this mod without shaders. But, even if this issue was closed, i'm posting it here because i don't feel like opening another thread describing the same exact issue.

**Edit: in fact, biomes with OptiFine HD U G5 alone seem to generate correctly. This bullshit happens only if you start Minecraft while you have one shader active. If you don't have shaders selected at Minecraft startup (have selected OFF or Internal) and select one later (regardless if you load the world or not), biomes generate correctly. It seems that it doesn't matter what shader you use, it has to be one random shader (even if i tested with BSL, Complementary and Sildur). Just wanted to update my reply here so mod authors and @ATLCode knows what it's going on and why there was a bit of confusion: ATLCode has probably started Minecraft with shaders and saw errors in worldgen, even if i'm not sure, and even if i'm not sure about an update i'm not aware of.**

**Long story short, latest Optifine is still kind of bugged with custom biome generation with these mods (happens with Biomes o'Plenty too but with the Nether biomes instead) but only if you start the game with a shader. Already went to the Optifine page to report this, i hope they fix this issue with shaders if possible. [This ](https://github.com/sp614x/optifine/issues/4880) is a similar issue but with 1.16.3 and where OP tells about the game crashing instead of biomes generating incorrectly. It's pretty old now but it seems that the latest Optifine didn't fully solve this issue**

**Comment by CynicalShrew** [comment]
> **Edit: in fact, biomes with OptiFine HD U G5 alone seem to generate correctly. This bullshit happens only if you start Minecraft while you have one shader active. If you don't have shaders selected at Minecraft startup (have selected OFF or Internal) and select one later (regardless if you load the world or not), biomes generate correctly. It seems that it doesn't matter what shader you use, it has to be one random shader (even if i tested with BSL, Complementary and Sildur). Just wanted to update my reply here so mod authors and @ATLCode knows what it's going on and why there was a bit of confusion: ATLCode has probably started Minecraft with shaders and saw errors in worldgen, even if i'm not sure, and even if i'm not sure about an update i'm not aware of.**
> 
> **Long story short, latest Optifine is still kind of bugged with custom biome generation with these mods (happens with Biomes o'Plenty too but with the Nether biomes instead) but only if you start the game with a shader. Already went to the Optifine page to report this, i hope they fix this issue with shaders if possible. [This ](https://github.com/sp614x/optifine/issues/4880) is a similar issue but with 1.16.3 and where OP tells about the game crashing instead of biomes generating incorrectly. It's pretty old now but it seems that the latest Optifine didn't fully solve this issue**

Can confirm, loading shaders after starting the game fixes this issue. Pain in the ass, though. It takes a solid two minutes for my shaders to finish loading, and I'm just using Sildur's Enhanced Default. On my Fabric instance, it takes maybe ten seconds. Hopefully the Optifine devs will put out an update to finally fix this mess.

**Comment by SatDog92** [comment]
> Can confirm, loading shaders after starting the game fixes this issue. Pain in the ass, though. It takes a solid two minutes for my shaders to finish loading, and I'm just using Sildur's Enhanced Default. On my Fabric instance, it takes maybe ten seconds. Hopefully the Optifine devs will put out an update to finally fix this mess.

Yeah but at last, there isn't much the mod authors can do here. I just wrote the reply so everyone gets where is the issue. Also, i feel your ass lmao, now i have a 32GB RAM PC with i9 and RTX 2060 card (mostly for photoshop, though), so loading shaders isn't an issue anymore for me, but back with my 8GB RAM one it was absolutely painful. And Fabric is generally faster than Forge (i do not know anything about Java coding, but wtf?)

**Comment by CorgiTaco** [comment]
This is an issue with ALL world gen mods. It is not specific to biome mod's. I'm also closing this thread as github issues are not a place for conversation.
